skymap_scanner/recos/millipede_wilks.py

```diff
@@ -44,16 +49,21 @@ class MillipedeWilks(RecoInterface):
     # At HESE energies, deposited light is dominated by the stochastic losses
     # (muon part emits so little light in comparison)
     # This is why we can use cascade tables
-    _splinedir = os.path.expandvars("$I3_DATA/photon-tables/splines")
-    _base = os.path.join(_splinedir, "cascade_single_spice_ftp-v1_flat_z20_a5.%s.fits")
-    _effd = os.path.join(_splinedir, "cascade_effectivedistance_spice_ftp-v1_z20.eff.fits")
-    for fname in [_base % "abs", _base % "prob", _effd]:
-        if not os.path.exists(fname):
-            raise FileNotFoundError(fname)
+    datastager = DataStager(
+        local_paths=cfg.LOCAL_DATA_SOURCES,
+        local_subdir=cfg.LOCAL_SPLINE_SUBDIR,
+        remote_path=f"{cfg.REMOTE_DATA_SOURCE}/{cfg.REMOTE_SPLINE_SUBDIR}",
+    )
+
+    datastager.stage_files(spline_requirements)
+
+    abs_spline = datastager.get_filepath(FTP_ABS_SPLINE)
+    prob_spline = datastager.get_filepath(FTP_PROB_SPLINE)
+    effd_spline = datastager.get_filepath(FTP_EFFD_SPLINE)
 
     cascade_service = photonics_service.I3PhotoSplineService(
-        _base % "abs", _base % "prob", timingSigma=0.0,
-        effectivedistancetable = _effd,
+        abs_spline, prob_spline, timingSigma=0.0,
+        effectivedistancetable = effd_spline,
         tiltTableDir = os.path.expandvars('$I3_BUILD/ice-models/resources/models/ICEMODEL/spice_ftp-v1/'),
         quantileEpsilon=1
         )
```

Comment:
Since this is much more dynamic now, it may be worth putting this into an `__init__()`. This way there are no surprises.

Reply:
Same goes for the other reco classes

Reply:
I think this was done to only load the tables once.  There may be a better way to do so (a singleton pattern?).

Reply:
@ric-evans I thought about that, but this would be a more significant change from the current design where everything in `RecoInterface` is static. I am just not sure whether the `traysegment` works when not defined as a static method.

Reply:
Okay, those are decent reasons to keep it as is. It's not pretty, but it's functional. I would like to see how the shared memory pattern works with repeated calls now.

Reply:
I am also not a big fan of the current design where we do not instantiate the `RecoInterface` class, but I think it also makes a little sense to have the `traysegment` as a method. Maybe it can refactored so that the traysegment function is built and returned by a method of the class itself. But I would consider this for a future PR.